### ISSUE TITLE
Build failure: emacs29-macport

### ISSUE THREAD

**Issue by sixtysecrun** [NixOS/nixpkgs#253502]
### Steps To Reproduce

1. build *emacs29-macport*

### Build log

```console
error: builder for '/nix/store/cawm1bdc0gb9gz2ya20za9fy2niz6jla-emacs-mac-macport-with-packages-29.1.drv' failed with exit code 1;
       last 1 log lines:
       > cp: cannot stat '/nix/store/63b32jd27qy6kz3mf3114w2qss66w3xy-emacs-mac-macport-29.1/Applications/Emacs.app/Contents/Info.plist': No such file or directory
       For full logs, run 'nix log /nix/store/cawm1bdc0gb9gz2ya20za9fy2niz6jla-emacs-mac-macport-with-packages-29.1.drv'.
```

### Additional context

The emacs is installed using home-manager on MacOS

```nix
{ config, pkgs, ... }:
let
  emacsPackage = pkgs.emacs29-macport;
in
{

  programs.emacs = {
    enable = true;
    package = emacsPackage;
    extraPackages = epkgs: [
      ...
    ];
  };
}
```

Reverting NixOS/nixpkgs@f6a6dc8 back to NixOS/nixpkgs@2160abc fixes the problem.

```nix
{ config, pkgs, ... }:
let
  workingPkgs = import (pkgs.fetchFromGitHub {
    owner = "NixOS";
    repo = "nixpkgs";
    rev = "2160abc3eddbe826b08bcffdb3758763b093f254";
    sha256 = "sha256-65gZ8Fa4EIMGOu5k0+dNiRi/9v2x+M9zDgly7fvDVa4=";
  }) {
    system = "x86_64-darwin";
  };
  emacsPackage = workingPkgs.emacs29-macport;
in
{

  programs.emacs = {
    enable = true;
    package = emacsPackage;
    extraPackages = epkgs: [
      ...
    ];
  };
}
```

### Notify maintainers

@AndersonTorres 

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-darwin"`
 - host os: `Darwin 22.6.0, macOS 10.16`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.17.0`
 - channels(user): `"darwin"`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

**Comment by AndersonTorres** [comment]
Well, I don't have machine to test Darwin things.

But, is this a home-manager related thing? Have you tried to run a nix-shell?

**Comment by sixtysecrun** [comment]
> is this a home-manager related thing?

I am not sure, but it surely stopped working after NixOS/nixpkgs@f6a6dc8 . 
Please also look at this file listings below 👇 

<details><summary>✅ Working Emacs.app</summary>
<p>

```console
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/bin
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/bin/ctags
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/bin/etags
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/bin/ebrowse
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/bin/emacs-29.1
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/bin/emacs
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/bin/emacsclient
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/MacOS
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/MacOS/Emacs
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/he.lproj
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/en_AU.lproj
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/ar.lproj
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/el.lproj
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/uk.lproj
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/English.lproj
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/English.lproj/InfoPlist.strings
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/es_419.lproj
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/es_MX.lproj
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/zh_CN.lproj
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/Dutch.lproj
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/pt_BR.lproj
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/da.lproj
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/document.icns
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/sk.lproj
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/pt_PT.lproj
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/German.lproj
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/ms.lproj
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/sv.lproj
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/cs.lproj
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/ko.lproj
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/no.lproj
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/hu.lproj
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/zh_HK.lproj
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/Spanish.lproj
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/tr.lproj
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/pl.lproj
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/zh_TW.lproj
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/en_GB.lproj
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/French.lproj
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/vi.lproj
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/ru.lproj
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/fr_CA.lproj
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/fi.lproj
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/id.lproj
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/th.lproj
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/pt.lproj
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/ro.lproj
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/Japanese.lproj
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/Emacs.icns
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/hr.lproj
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/hi.lproj
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/Italian.lproj
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Resources/ca.lproj
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/Info.plist
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/Applications/Emacs.app/Contents/PkgInfo
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/share
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/share/man
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/share/info
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/share/icons
/nix/store/4i36ghfh9cwgrasx7923zx1s7lwf61mp-emacs-mac-macport-with-packages-29.1/share/applications
```

</p>
</details> 




<details><summary>❌ Latest Emacs.app</summary>
<p>

```console
nix build nixpkgs#emacs29-macport
```


```console
> find result/Applications/Emacs.app/

result/Applications/Emacs.app/
result/Applications/Emacs.app/Contents
result/Applications/Emacs.app/Contents/MacOS
result/Applications/Emacs.app/Contents/MacOS/Emacs.sh
result/Applications/Emacs.app/Contents/Resources
result/Applications/Emacs.app/Contents/Resources/document.icns
result/Applications/Emacs.app/Contents/Resources/Emacs.icns
result/Applications/Emacs.app/Contents/native-lisp
result/Applications/Emacs.app/Contents/PkgInfo
```

</p>
</details> 

It seems that there is fewer files now in the output for whatever reason?

**Comment by knl** [comment]
I'm having the same issue on darwin with emacs28-macport

**Comment by knl** [comment]
Running `git bisect run sh -c 'nix-build -A emacs28-macport && fd . result/Applications/Emacs.app |rg Info.plist` between the `HEAD` and ae4e2e8676af606cb5c38f9f51e21f6053e41f10 yields this as the breaking commit:

```
running 'sh' '-c' 'nix-build -A emacs28-macport && fd . result/Applications/Emacs.app |rg Info.plist'
/nix/store/jkdbjl02k2k45gp4kkqjg3cgvalrbkrp-emacs-mac-macport-28.2
f607a7851619a6efda5f7c57f0345018a3ea403c is the first bad commit
commit f607a7851619a6efda5f7c57f0345018a3ea403c
Author: Andrew Pan <a@tny.town>
Date:   Tue Aug 29 12:52:53 2023 -0500

    emacs-macport: build on LLVM 14

    - Build on `apple_sdk_11_0` to support `aarch64-darwin`
    - Add frameworks required for building on newer SDKs
      - `Accelerate`, `UniformTypeIdentifiers`
    - Include header to work around `CF_NOESCAPE` issue circa LLVM 7.0

 pkgs/applications/editors/emacs/default.nix        |  6 ++---
 pkgs/applications/editors/emacs/generic.nix        | 13 +++++++--
 .../editors/emacs/macport_noescape_noop.h          | 31 ++++++++++++++++++++++
 pkgs/top-level/all-packages.nix                    |  2 +-
 4 files changed, 46 insertions(+), 6 deletions(-)
 create mode 100644 pkgs/applications/editors/emacs/macport_noescape_noop.h
bisect found first bad commit
```

**Comment by AndersonTorres** [comment]
Invoking @tnytown

**Comment by wegank** [comment]
Closed by #253041. Also built on Hydra: https://hydra.nixos.org/job/nixpkgs/trunk/emacs29-macport.x86_64-darwin

**Comment by knl** [comment]
@wegank I think the issue we're concerned about is still not fixed. `Info.plist` is missing and `emacs{28,29}-macport` is incorrectly built.
